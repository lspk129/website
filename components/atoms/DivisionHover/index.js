import styled from 'styled-components';

const DivisionHover = styled.div`
  transition: all 0.5s ease-in-out;
  ${({ height }) => height && `height: ${height}px`};
  ${({ scale }) => scale && `
    img {
      width: 200px;
      height: 200px;
    }
    &:hover {
      transform: scale(1.1);
  }

  `}
  ${({ bgcolor, theme }) => bgcolor && `
    &:hover {
      background: ${theme.background};
      color: ${theme.text};
      .icons {
        color: ${theme.text};
      }
    };
    .icons {
      transition: all 0.5s ease-in-out;
      color: ${theme.background};
    }
  `}
`;

export default DivisionHover;
