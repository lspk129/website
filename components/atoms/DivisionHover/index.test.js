import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import DivisionHover from './';

describe('DivisionHover', () => {
  const shallowWrapper = shallow(<DivisionHover />);

  it('renders correctly', () => {
    const wrapper = renderer.create(<DivisionHover />);
    const tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders height props', () => {
    const wrapper = renderer.create(<DivisionHover height={100} />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('renders scale boolean props', () => {
    const wrapper = renderer.create(<DivisionHover scale />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('renders background color boolean props with theme props', () => {
    const wrapper = renderer.create(<DivisionHover bgcolor />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('renders div tag', () => {
    expect(shallowWrapper).toBePresent();
    expect(shallowWrapper).toHaveTagName('div');
  });
});
