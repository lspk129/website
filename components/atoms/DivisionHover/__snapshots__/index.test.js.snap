// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DivisionHover renders background color boolean props with theme props 1`] = `
<div
  className="sc-bdVaJa ialuuP"
/>
`;

exports[`DivisionHover renders correctly 1`] = `
<div
  className="sc-bdVaJa gTxdXK"
/>
`;

exports[`DivisionHover renders height props 1`] = `
<div
  className="sc-bdVaJa cvWOwL"
  height={100}
/>
`;

exports[`DivisionHover renders scale boolean props 1`] = `
<div
  className="sc-bdVaJa liLKLb"
  scale={true}
/>
`;
